maestro-app/src/main/java/bio/overture/maestro/app/infra/adapter/inbound/messaging/IndexingMessagesStreamListener.java

```diff
@@ -1,107 +1,120 @@
+            .subscribe(tuple -> log.info(" processed message : {} success : {}", tuple._1(), tuple._2().isSuccessful()));
     }
 
     @StreamListener
-    public void handleIndexStudyMessage(@Input(Sink.INPUT) Flux<IndexStudyMessage> indexStudyMessageFlux) {
-        indexStudyMessageFlux.subscribe(msg -> {
-            try {
-                indexer.indexStudy(IndexStudyCommand.builder()
+    public void handleIndexRepositoryMessage(@Input(Sink.INPUT) Flux<IndexRepositoryMessage> indexRepoMessageFlux) {
+        indexRepoMessageFlux.flatMap(msg-> this.doCheckedCall(msg, this::indexRepository))
+            .subscribe(tuple -> log.info(" processed message : {} success : {}", tuple._1(), tuple._2().isSuccessful()));
+    }
+
+    @NotNull
+    private Mono<Tuple2<IndexAnalysisMessage, IndexResult>> indexOrRemoveAnalysis(IndexAnalysisMessage msg) {
+        if (msg.getRemove()) {
+            return removeAnalysis(msg);
+        } else {
+            return indexAnalysis(msg);
+        }
+    }
+
+    @NotNull
+    private Mono<Tuple2<IndexAnalysisMessage, IndexResult>> removeAnalysis(IndexAnalysisMessage msg) {
+        return indexer.removeAnalysis(RemoveAnalysisCommand.builder()
+            .analysisIdentifier(AnalysisIdentifier.builder()
                     .studyId(msg.getStudyId())
+                    .analysisId(msg.getAnalysisId())
                     .repositoryCode(msg.getRepositoryCode())
-                    .build()
-                ).onErrorResume((e) -> {
-                    log.error("failed reading message: {} ", msg, e);
-                    return Mono.empty();
-                }).subscribe(indexResult ->
-                    log.info(" processed message : {} success : {}", msg, indexResult.isSuccessful())
-                );
-            } catch (Exception e) {
-                log.error("failed reading message: {} ", msg, e);
-            }
-        });
+                    .build())
+                .build())
+            .map(out -> new Tuple2<>(msg, out))
+            .onErrorResume((e) -> catchUnhandledErrors(msg, e));
     }
 
-    @StreamListener
-    public void handleIndexRepositoryMessage(@Input(Sink.INPUT) Flux<IndexRepositoryMessage> indexRepoMessageFlux) {
-        indexRepoMessageFlux.subscribe( msg -> {
-            try {
-                indexer.indexStudyRepository(IndexStudyRepositoryCommand.builder()
-                    .repositoryCode(msg.getRepositoryCode())
-                    .build()
-                ).onErrorResume((e) -> {
-                    log.error("failed reading message: {} ", msg, e);
-                    return Mono.empty();
-                }).subscribe(indexResult ->
-                    log.info(" processed message : {} success : {}", msg, indexResult.isSuccessful())
-                );
-            } catch (Exception e) {
-                log.error("failed reading message: {} ", msg, e);
-            }
-        });
+    @NotNull
```

Comment:
this is a jetbrains dependency, not sure if this was planned or a mistake. according to https://www.jetbrains.com/help/idea/nullable-and-notnull-annotations.html#notnull it will catch non null returned objects.

Reply:
by mistake removing, they add it when i refactor methods